src/telemetry.ts

```diff
@@ -37,8 +50,57 @@ export class Telemetry {
 
   async sendEvent(event: TelemetryEvent) {
     if (await this.initialize()) {
-      return this.api!.sendEvent(event);
+      this.api!.sendEvent(event);
+    }
+  }
+
+  async sendConfigurationEvents() {
+    const lastConfigurationTelemetry: number | undefined =
+      this.context.globalState.get("rubyLsp.lastConfigurationTelemetry");
+
+    if (
+      lastConfigurationTelemetry &&
+      Date.now() - lastConfigurationTelemetry <= ONE_DAY_IN_MS
+    ) {
+      return;
     }
+
+    const promises: Promise<void>[] = [
+      { namespace: "workbench", field: "colorTheme" },
+      { namespace: "rubyLsp", field: "enableExperimentalFeatures" },
+      { namespace: "rubyLsp", field: "yjit" },
+      { namespace: "rubyLsp", field: "rubyVersionManager" },
+      { namespace: "rubyLsp", field: "formatter" },
+    ].map(({ namespace, field }) => {
+      return this.sendEvent({
+        namespace,
+        field,
+        value: (
+          vscode.workspace.getConfiguration(namespace).get(field) ?? ""
+        ).toString(),
+      });
+    });
+
+    const enabledFeatures = vscode.workspace
+      .getConfiguration("rubyLsp")
+      .get("enabledFeatures")!;
```

Comment:
Isn't there a chance `get("enabledFeatures")` could be null?

Reply:
I don't think so because we have a default value configured. It always returns an object.